sulong/projects/com.oracle.truffle.llvm.libraries.bitcode/src/pthreads.c

```diff
@@ -27,15 +27,40 @@
  * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
  * OF THE POSSIBILITY OF SUCH DAMAGE.
  */
+#include <errno.h>
 #include <sys/types.h>
 #include <unistd.h>
 #include <pthread.h>
 
 #include "unsupported.h"
 
-int pthread_attr_destroy(pthread_attr_t *attr) {
-  ERR_UNSUPPORTED(pthread_attr_destroy);
+int __sulong_getEBUSY() {
+	return EBUSY;
 }
+int __sulong_getEINVAL() {
```

Comment:
EINVAL and friends are already defined as constants in `com.oracle.truffle.llvm.runtime.nodes.asm.syscall.LLVMAMD64Error`. Your solution is more versatile in that it imports them from the actual native header. However, this will not work in managed mode, where access to native functions is prohibited even for internal functions like these. Please change your code to use the already hardcoded constant values, and remove the `UtilCConstants` class.

Reply:
ok